ocaml/testsuite/tests/parsing/pr6604.compilers.reference

```diff
@@ -1,4 +1,4 @@
 File "pr6604.ml", line 9, characters 0-2:
 9 | #1
     ^^
-Error: Syntax error: line number directive not expected.
+Error: Syntax error: unboxed integer literal with type-specifying suffix expected.
```

Comment:
Unfortunately, this error is issued not only at the beginning of lines, but wherever this appears.

I think adding a hint about line number directives is confusing for the case where someone just forgets a suffix on their unboxed int literal (which, in the not-too-distant future will hopefully be much more common than line number directives).

It might conceivably be possible to distinguish the beginning of line case, but I think not without some lexer trickery, and propose it is not worth investing further time in right now.

I have updated the error along the lines you suggest, though.